src/components/common/data/DataView.tsx

```diff
@@ -1,46 +1,81 @@
 import React, { Fragment } from 'react';
-import PropTypes from 'prop-types';
-import { withStyles } from '@mui/styles';
+import { ClassNameMap, withStyles } from '@mui/styles';
 import { Button, Grid, Typography } from '@mui/material';
+import { CustomTheme } from '@mui/material/styles';
+import { connect } from 'react-redux';
 
 import { Link } from 'react-router-dom';
+import LightTable from 'components/table/LightTable';
 import SidebarDrawer from 'components/dataset/data/sidebar/SidebarDrawer';
 import DataViewDropdown from './DataViewDropdown';
-import JadeTable from '../../table/JadeTable';
 import SnapshotPopup from '../../snapshot/SnapshotPopup';
 import AppBreadcrumbs from '../../AppBreadcrumbs/AppBreadcrumbs';
 import { BreadcrumbType } from '../../../constants';
+import { TableColumnType, TableRowType } from '../../../reducers/query';
+import { TdrState } from '../../../reducers';
 
-const styles = (theme) => ({
+const styles = (theme: CustomTheme) => ({
   pageRoot: { ...theme.mixins.pageRoot },
   pageTitle: { ...theme.mixins.pageTitle },
   wrapper: {
     paddingTop: theme.spacing(0),
     padding: theme.spacing(4),
   },
   scrollTable: {
-    overflowY: 'auto',
+    overflow: 'auto',
     height: '100%',
+    paddingTop: theme.spacing(1),
   },
 });
 
+type DataViewProps = {
+  canLink: boolean;
+  classes: ClassNameMap;
+  columns: Array<TableColumnType>;
+  filterStatement: string;
+  handleChangeTable: () => void;
+  handleDrawerWidth: () => void;
+  pageBQQuery: () => void;
+  panels: Array<object>;
+  polling: boolean;
+  resourceId: string;
+  resourceLoaded: boolean;
+  resourceName: string;
+  resourceType: string;
+  rows: Array<TableRowType>;
+  selected: boolean;
+  selectedTable: string;
+  sidebarWidth: number;
+  tableNames: Array<string>;
+  totalRows: number;
+};
+
 function DataView({
   canLink,
   classes,
+  columns,
+  filterStatement,
   handleChangeTable,
   handleDrawerWidth,
   pageBQQuery,
   panels,
-  queryParams,
+  polling,
   resourceId,
   resourceLoaded,
   resourceName,
   resourceType,
+  rows,
   selected,
   selectedTable,
   sidebarWidth,
   tableNames,
-}) {
+  totalRows,
+}: DataViewProps) {
+  const rowKey = (row: TableRowType): string => {
+    const drId = row.datarepo_row_id;
+    return drId ?? row.name;
```

Comment:
huh...wasn't familiar with this syntax.  What does that do?

Reply:
> The nullish coalescing operator (??) is a logical operator that returns its right-hand side operand when its left-hand side operand is null or undefined, and otherwise returns its left-hand side operand.

For the non-data view tables, datarepo_row_id will be null, so we can just use the row name.

Reply:
oh cool.  it's like a coalesce?

Reply:
I use it as a short-hand for a ternary statement - the above is equivalent to`drId != null ? drId : row.name`.